### ISSUE TITLE
New rule proposal: max-accessor-depth

### ISSUE THREAD

**Issue by seafoam6** [eslint/eslint#11779]
**Please describe what the rule should do:**
Should error if a property reference is too deep. `a.b.c.d.e` vs `a.b`. 

**What category of rule is this? (place an "X" next to just one item)**

[ X ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
"max-accessor-depth": [
    2, 3
]
```

**Why should this rule be included in ESLint (instead of a plugin)?**
I think everyone in their JS career has hit the "cannot find property of undefined" at runtime. 

**Are you willing to submit a pull request to implement this rule?**
Yes!

**Comment by kaicataldo** [comment]
Hi, thanks for the issue. Do you mind updating your proposal with some more code examples and how configuration would affect them? It would be helpful to see a few different configurations and what would be correct/incorrect code when the rule is configured as such. I'm assuming that in the example configuration (`[2, 3]`), `2` refers to the warning level while `3` refers to the maximum depth of properties one can reach for, but it's not entirely clear to me.

Also, I'm not entirely sure how this would directly help solve the problem of unexpected undefined values. Do you mind adding some additional clarifying information about the problem you're trying to solve and how you think this solution solves it?

Given the sheer volume of rules we already have in core, we have a really high bar for adding new rules. This additional information will help us decide if we think we should add this as part of core or if it maybe belongs in a custom rule.

**Comment by ljharb** [comment]
what about:
```js
const { b: { c: { d: { e } } } } = a;
```
or
```js
const { b } = a;
const { c } = b;
const { d } = c;
const { e } = d;
```

Can you elaborate on why this is valuable? The depth doesn't really correlate to the likelihood of an undefined property.

**Comment by seafoam6** [comment]
I'm going to close this out for now. I'm going to see if I can get it to work in a plugin before proposing. 

Hadn't even considered `const { b: { c: { d: { e } } } } = a;`
